### DISCUSSION TITLE
Web-Based Dashboard

### DISCUSSION THREAD

**Discussion by ArcaneDisgea** [the-convocation/venat#89] [Ideas]
This dashboard should come in two parts. The core bot maintenance and configuration panel and the per-guild plugin configuration panels.

I really have not many thought here yet but want to get the discussion down and rolling since we are talking about it in #88.

**Comment by karashiiro** [comment]
We could probably just have an application shell with navbar sections, and the config/global settings are only visible to bot admins. Config itself can just be a key-value thing like Vercel's environment variable configuration. I don't think this will end up being very complicated to start with. One thing to keep in mind is that if we want to allow plogons to have their own UIs, we'll need a way for those to be mounted to the website. We could do with a polyglot framework or something.

**Comment by karashiiro** [comment]
If we wanted to dodge microfrontends (they cause their own set of problems), we could choose a framework and define a module that abstracts away as much of the framework as possible (including lifecycle and event hooks) and provide a set of components for that.

**Comment by karashiiro** [comment]
@ackwell should have a closer read on our options.
